packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Models/Serialization/SerializationInterfaces.cs

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Microsoft.Generator.CSharp
+{
+    public abstract class SerializationInterfaces : IEnumerable<CSharpType>
+    {
+        protected SerializationInterfaces(TypeProvider typeProvider, bool hasJson, bool hasXml) { }
+
+        public CSharpType? IJsonModelTInterface { get; init; }
```

Comment:
These properties are interesting - do we need the ability for overriders to specify their own versions of each of these interfaces?

Reply:
Resolving based on our internal discussion